### PR TITLE
Doxygen for isrchne.f iw3jdn.f iw3mat.f iw3pds.f iw3unp29.f (Part of #77)

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
Doxygen for:
- isrchne.f
- iw3jdn.f
- iw3mat.f
- iw3pds.f
- iw3unp29.f

Part of #77

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
@Hang-Lei-NOAA I'd like extra help with iw3unp29.f because it's a large file and i haven't dealt with multiple functions in one file yet.

**Comment by Hang-Lei-NOAA** [comment]
Okay, I will deal with that file, and then send it to you.

**Comment by BrianCurtis-NOAA** [comment]
> Okay, I will deal with that file, and then send it to you.

Hang,

Have you had a chance to look this file over? Thanks!

**Comment by Hang-Lei-NOAA** [comment]
I have finished the major part. But there are too many functions without
any reference. I have to read the code and then write the doxygen docs.

File: src/iw3unp29.f [hunk 1]
```diff
@@ -738,11 +727,11 @@ FUNCTION I03O29(NUNIT, OBS, IER)
          REWIND NUNIT
          ISWT = 0
       END IF
- 
+
    10 CONTINUE
 
       IF(NEXT.NE.0)  GO TO 70
- 
+
```
Comment (edwardhartnett) [inline src/iw3unp29.f:741]: Whatever it is you're doing to remove these spaces, if should be done as a separate step from the doxygenation.

That is, do the doxygen without touching these lines, to make review easier, then do another pass when you remove trailing blanks (which I assume is being done by your editor...)

File: src/iw3unp29.f [hunk 2]
```diff
@@ -2083,7 +2072,7 @@ FUNCTION L01O29()
 C***********************************************************************
       FUNCTION R03O29(LUNIT,OBS)
```
Comment (edwardhartnett) [inline src/iw3unp29.f:2092]: OK, these are the functions that @Hang-Lei-NOAA is mentioning. How many are there?

**Comment by Hang-Lei-NOAA** [comment]
Got it.

File: src/iw3unp29.f [hunk 3]
```diff
@@ -4569,20 +4554,17 @@ SUBROUTINE S06O29(IDEN,ID)
       RETURN
       END
 
+C> @brief Finds the location of the next numeric character
```
Comment (edwardhartnett) [inline src/iw3unp29.f]: If this is a subroutine doxygen block, it should not have @brief, and should not be separated by a blank line from the params. We only need one file-level block, at the top of the file...

**Comment by Hang-Lei-NOAA** [comment]
This has been corrected in my end. Please wait for a while for review, I am
finishing the rest.

File: src/iw3unp29.f (hunk 1 above)
Comment (BrianCurtis-NOAA) [reply src/iw3unp29.f:741]: I have no idea what's causing it yet. I am using atom.io but it doesn't happen with all the files I edit, which makes it weird. As soon as I figure out what is causing it, I intend to remedy it.

Comment (edwardhartnett) [reply src/iw3unp29.f:741]: OK. ;-) Sometimes editors can be funny!

**Comment by Hang-Lei-NOAA** [comment]
@Brian Curtis - NOAA Affiliate ***@***.***>
I made some major edits and pass to you.
Please combine them first.
I have to be hurry on another task associated with the hpc-stack v1.2.0
release, which is due this afternoon.
There are still many (20+) functions are missing docs, we can work on that
later.
Please access it on jet:
/mnt/lfs4/HFIP/hfv3gfs/Hang.Lei/nceplibs/docus/iw3unp29.f

**Comment by edwardhartnett** [comment]
I agree with @Hang-Lei-NOAA that if you want to merge in these changes, and circle back around to the functions missing doxygen, that's OK...

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Pull request merged**